ynr/apps/sopn_parsing/helpers/pdf_helpers.py

```diff
@@ -69,7 +69,19 @@ def match_ballot_to_pages(self, ballot: Ballot) -> str:
                     page_numbers.append(page.page_number)
                     page.matched = True
                     previous_page = page
-        return ",".join([str(page_num) for page_num in page_numbers])
+            if self.contains_consecutive_pages():
+                return ",".join([str(page_num) for page_num in page_numbers])
+            else:
+                raise ValueError("Page numbers are not consecutive")
+
+    @property
+    def contains_consecutive_pages(self):
```

Comment:
This needs to check the matched pages are consecutive (e.g page_numbers). The document pages are always consecutive by definition, as we assign page numbers in order.

contains_consecutive_pages should accept a list as an arg and check that list, but the logic above can stay more or less as is


(left a previous comment on a now missing commit, sorry for the duplicate!)

Reply:
Of course 🤦‍♀️ 

I thought pushing with `--force-with-lease` would save the comments :/